operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +78,207 @@ type MachineConfigurationList struct {
+	// +optional
+	ClusterDefaultPolicies []NodeDisruptionPolicyConfig `json:"clusterDefaultPolicies"`
+}
+
+type NodeDisruptionPolicyStatus struct {
+	// clusterPolicies is a merge of cluster default and user provided node disruption policies.
+	// +optional
+	ClusterPolicies []NodeDisruptionPolicyConfig `json:"clusterPolicies"`
+
+	// validation of the provided spec, one of ('Success', 'Failed').
+	// +optional
+	Validation Validation `json:"validation"`
+
+	// reason is a human readable explanation for the failure of policy validation, if any
+	// +optional
+	Reason string `json:"reason"`
+}
+
+// TODO is a union discriminator valid for type/value if I have Actions as well?
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'sshkey' ? has(self.value) : has(self.value)",message="Setting a value is forbidden for specified type"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type != 'sshkey' ? has(self.value) : !has(self.value)",message="Setting a value is required for specified type"
+type NodeDisruptionPolicyConfig struct {
+	// type represents the type of the object that this policy is in effect for, matching available MachineConfig
+	// fields (files, services, kernel arguments, and sshkeys)
+	// +kubebuilder:validation:Required
+	Type NodeDisruptionPolicyType `json:"type"`
+	// value represents the canoncial name of the object that this policy is in effect for.
+	// This could be a file path, a service name, the specific kernel argument, etc.
+	// +optional
+	Value string `json:"value"`
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// Validation is a string enum used in a NodeDisruptionPolicyStatus object. They describe whether the controller
+// was able to validate the latest spec
+// +kubebuilder:validation:Enum:="success";"failed"
+type Validation string
+
+const (
+	// success represents a valid cluster node disruption policy
+	Success Validation = "success"
+
+	// failed represents an invalid cluster node disruption policy
+	Failed Validation = "failed"
+)
+
+// NodeDisruptionPolicyType is a string enum used in a NodeDisruptionPolicyConfig object. They describe the type of
+// the object that this policy will be in effect for.
+// +kubebuilder:validation:Enum:="file";"service";"kernelArgument";"sshKey"
+type NodeDisruptionPolicyType string
+
+const (
+	// File represents a NodeDisruption policy that is in effect for changes to a file.
+	File NodeDisruptionPolicyType = "file"
+
+	// Service represents a NodeDisruption policy that is in effect for changes to a service.
+	Service NodeDisruptionPolicyType = "service"
+
+	// File represents a NodeDisruption policy that is in effect for changes to a kernel argument.
+	KernelArgument NodeDisruptionPolicyType = "kernelArgument"
```

Comment:
MCO doesn't have a way today to apply kargs without reboot. Are we thinking of adding a way to apply these kargs live? If not, perhaps we may want to skip letting user skip drain/reboot in the initial implementation.